server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Scanner.java

```diff
@@ -38,100 +40,126 @@
   private ScanDataSource isolatedDataSource;
   private boolean sawException = false;
   private boolean scanClosed = false;
+  protected Semaphore scannerSemaphore;
 
   Scanner(Tablet tablet, Range range, ScanOptions options) {
     this.tablet = tablet;
     this.range = range;
     this.options = options;
+    scannerSemaphore = new Semaphore(1, true);
```

Comment:
The expense of fairness may not be a problem.  I suspect the expense of fairness is an issue for semaphores that are frequently accessed by many threads, which is not the case here.